climada/engine/supplychain.py

```diff
@@ -0,0 +1,316 @@
+            year (int): Year of wiot table. Valid years go from 2000 to 2014.
+                        Default 2014.
+            file_path (str): Path to folder where the wiod table is stored.
+                            Deafult is SYSTEM_DIR. If data are not present, they
+                            will be downloaded in save_dir, i.e., ~/climada/data/results
+                            If user-defined, user_data must be set to True
+            Default values of the last four args allow reading the full wiot table.
+        """
+
+        file_name = 'WIOT{}_Nov16_ROW.xlsb'.format(year)
+        file_folder = Path(file_folder)
+        file_folder.mkdir(exist_ok=True)
+        file_loc = file_folder / file_name
+
+        if not file_loc in file_folder.iterdir():
+            download_link = WIOD_FILE_LINK + file_name
+            u_fh.download_file(download_link, download_dir=file_folder)
+            LOGGER.info('Downloading WIOD table for year %s', year)
+        mriot = pd.read_excel(file_loc, engine='pyxlsb')
+
+        # hard-coded values based on the structure of the wiod tables
+        col_sectors = 1
+        col_iso3 = 2
+        end_row_sectors = 61
+        start_row, end_row = (5, 2469)
+        start_col, end_col = (4, 2468)
+
+        self.sectors = mriot.iloc[start_row:end_row_sectors, col_sectors].values
+        self.countries_iso3 = mriot.iloc[start_row:end_row, col_iso3].unique()
+        self.mriot_data = mriot.iloc[start_row:end_row,
+                                     start_col:end_col].values
+        self.total_prod = mriot.iloc[start_row:end_row, -1].values
+        self.cntry_pos = {
+            iso3: range(len(self.sectors)*i, len(self.sectors)*(i+1))
+            for i, iso3 in enumerate(self.countries_iso3)
+            }
+        self.mriot_type = 'wiod'
+
+    def calc_sector_direct_impact(self, hazard, exposure, imp_fun_set,
+                                  sector_type='service', selected_subsec=None):
+        """Calculate for each country/sector-combination the direct impact per year.
+        I.e. compute one year impact set for each country/sector combination. Returns
+        the notion of a supplychain year impact set, which is a dataframe with size
+        (n years) * ((n countries)*(n sectors)).
+
+        Parameters:
+            hazard (Hazard): Hazard object for impact calculation.
+            exposure (Exposures): Exposures object for impact calculation.
+            imp_fun_set (ImpactFuncSet): Set of impact functions.
+            selected_subsec (list): User-defined list with positions of the
+                                    subsector to analyze. Default None.
+            sector_type (str): If sec_subsec is not defined; it sets the start
+                and end positions in the mriot tablefor some default sectors.
+                Possible values are "service", "manufacturing", "agriculture"
+                and "mining". Either sec_subsec or sector_type must be defined.
```

Comment:
Question: what happens when both `sector_type` and `selected_subsec` are set?

Suggestion: make only one variable which is either a list with positions or a string. although it is slightly transvestite in types, this is not unseen in Python and could simplify the whole thing. 
```suggestion
            selected_subsec : list(int) or string
                Selection of sectors/subsectors to aggregate. It is either 1) a list of the index positions of the
                selected sectors/subsectors in the WIOT table, or 2) one of the pre-defined list of subsectors
                "service", "manufacturing", "agriculture", or "mining" for the WIOD data.
                 The default is "service".
```

Reply:
done